### ISSUE TITLE
Unexpected error using Field Components and grouped checkboxes

### ISSUE THREAD

**Issue by Josh-Nicholson** [ciscoheat/sveltekit-superforms#165]
**Description**

An unexpected svelte error occurs in a form where the following happens:
1. The form is using generic field components (like TextField.svelte in the docs). This means that +page.svelte retrieves/initialises the form like ``` const form = superForm(data.form); ``` rather than ``` const {form} = superForm(data.form) ```
2. The form uses grouped checkboxes that is mapped to schema property that is a string array (like in this [Ice Cream Demo](https://stackblitz.com/edit/sveltekit-superforms-group-inputs?file=src%2Froutes%2F%2Bpage.server.ts,src%2Froutes%2F%2Bpage.svelte))
3. You get the grouped checkboxes value and errors using ```const { value, errors } = formFieldProxy(form, 'exampleStringArray'); ```
4. Within the form you output the checkbox errors using 
``` svelte
 {#if $exampleStringArray}
    {#each $exampleStringArray as error}
      <label for="exampleStringArray" class="label py-0">
        <div class="flex text-red-500 font-medium">{error}</div>
      </label>
    {/each}
  {/if}
```
5. When more than one of the grouped checkboxes gets checked the following error happens
![image](https://github.com/ciscoheat/sveltekit-superforms/assets/9265897/5e21428c-9079-4e09-98a8-31375ad3e248)

Here is a link to repo with a full example:
[Gitub repo](https://github.com/Josh-Nicholson/SveltekitSuperformsErrorExample)

**Comment by ciscoheat** [comment]
The simple answer is that you cannot have proxy for an array. A warning will be issued for this in v1.0, but for now, you just have to do it some other way, unfortunately.

**Comment by Josh-Nicholson** [comment]
So from my testing the value proxy works with an array, just not the error proxy.
For the errors I guess I can set an error at form level and display that.

**Comment by Josh-Nicholson** [comment]
I've refactored the form to not use formFieldProxy's and I still get the same error.

Displaying errors with the following code still gives the svelte each error
``` svelte
{#if $errors.roles}
  <div class="flex flex-col">
  {#each $errors.roles as error}
	  <span class="flex text-red-500 font-medium">{error}</span>
  {/each}
  </div>
{/if}
```

However display them like this outputs [Object object]
``` svelte
{#if $errors.roles}<span class="flex text-red-500 font-medium">{$errors.roles}</span>{/if}
```

Any idea on why this is happening?

**Comment by Josh-Nicholson** [comment]
Outputting the [Object object] to the console you get

![image](https://github.com/ciscoheat/sveltekit-superforms/assets/9265897/6f3d33fb-80d0-4859-8a68-fffe56e0ea0f)

This is strange as firstly I'm not expecting an error to occur at all.
Secondly if there is an error why are the object properties undefined?

**Comment by Josh-Nicholson** [comment]
Here is a slightly tweaked ice cream demo on stackblitz demonstrating the issue

[Example](https://stackblitz.com/edit/sveltekit-superforms-group-inputs-s4op5q?file=src%2Froutes%2F%2Bpage.svelte)

**Comment by Josh-Nicholson** [comment]
Using [this previusly raised issue](https://github.com/ciscoheat/sveltekit-superforms/issues/138) I was able to display the array errors correctly

```svelte
{#if $errors.roles}
  {#each Object.entries($errors.roles) as error, idx}
    {#if error[1]}
	<span class="flex text-red-500 font-medium">{error[1]}</span>
    {/if}
  {/each}
{/if}
```

Still slightly confused as to why the errors.roles property has this structure when the schema only generates one error message for the whole array, not individual elements.

Would be good if there was a quicker/cleaner way to access errors on the whole property not the individual array elements

**Comment by ciscoheat** [comment]
Adding errors to an array or object in the schema is not supported. If you look at the schema as a tree, the nodes cannot have any errors, only the leaves, since they have to match the structure of the schema.

```ts
const schema = z.object({
  roles: z.string().array()
})
```

```
Schema:
 root
     \
     roles
    /     \
  [0]     [1]
string   string

Errors:
 root
     \
     roles <-- Cannot have errors, since it would then be a `string[]`
    /     \
  [0]     [1]
string[] string[]
```

**Comment by ciscoheat** [comment]
Some good news for 1.0, already available in 1.0.0.-rc.3: Now you can handle errors for arrays in the schema, like a minimum or maximum number of items:

```ts
const schema = z.object({
  tags: z.string().array().max(3)
})

const { form, errors } = superForm(data.form);
```

```svelte
{#if $errors.tags._errors}
  {$errors.tags._errors}
{/if}
```
